### PR TITLE
nversion: read version from text file

### PR DESCRIPTION
**Pull request by alaviss**
## Summary
Instead of statically encoding the version as a structure, read it from
a text file. This allows tools to modify the version string without
creating invalid NimSkull, and allow for easier rearrangement of the
code in the future.

## Details
* Added validation for version string suffix according to SemVer rules.
* Compiler (base) version is now stored in text form in 
`compiler/version.txt` .

### CODE REVIEW

File: compiler/utils/nversion.nim [hunk 1]
```diff
@@ -32,19 +43,24 @@ func parse*(s: string): Version =
...
     result.major, result.minor, result.patch, result.suffix
-  ):
+  ) or not result.suffix.isValidSuffix:
     raise newException(ValueError):
-      "Invalid version string: " & s
+      "Invalid version string: " & s.escape()
+
+const CompilerVersionSuffix* {.strdefine.} = ""
+  ## The suffix to attach to the compiler version. This is meant to be
+  ## declared by build tools to signify development version for example.
```
Comment (zerbina) [inline compiler/utils/nversion.nim:52]:
```suggestion
const
  CompilerVersionSuffix* {.strdefine.} = ""
    ## The suffix to attach to the compiler version. This is meant to be
    ## declared by build tools to signify development version for example.
```
Very minor thing that doesn't necessarily have to change. Using a block from the start makes adding more constants to it easier in the future.

**Review by zerbina — APPROVED** [review]
Looks good to me. I left one small, optional suggestion to use a `const` block.

File: compiler/utils/nversion.nim (hunk 1 above)
Comment (alaviss) [reply compiler/utils/nversion.nim:52]: I'd rather that new constants be put at the end of the file like its currently is :P

**Pull request merged**